UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/Matrix.cs

```diff
@@ -267,17 +267,17 @@ public bool IsNoGravityAt(Vector3Int position, bool isServer)
 		return MetaTileMap.IsNoGravityAt(position, isServer);
 	}
 
-	public IEnumerable<RegisterTile> GetRegisterTile(Vector3Int localPosition, bool isServer)
+	public List<RegisterTile> GetRegisterTile(Vector3Int localPosition, bool isServer)
 	{
 		return (isServer ? ServerObjects : ClientObjects).Get(localPosition);
 	}
 
 	//Has to inherit from register tile
-	public IEnumerable<T> GetAs<T>(Vector3Int localPosition, bool isServer) where T : RegisterTile
+	public List<T> GetAs<T>(Vector3Int localPosition, bool isServer) where T : RegisterTile
 	{
 		if (!(isServer ? ServerObjects : ClientObjects).HasObjects(localPosition))
 		{
-			return Enumerable.Empty<T>(); //?
+			return new List<T>(); //?
```

Comment:
I think `Enumerable.Empty<T>` doesnt GC, while new `List<T>` does.

https://stackoverflow.com/a/34786905

Reply:
what solution would be best, since there are a few places that do to list

Reply:
well, there's places that return lists instead of  Enumerable, so they have to do to list